extensions/ql-vscode/src/run-queries.ts

```diff
@@ -169,6 +171,29 @@ export class QueryInfo {
   async hasInterpretedResults(): Promise<boolean> {
     return fs.pathExists(this.resultsPaths.interpretedResultsPath);
   }
+
+  /**
+   * Holds if this query already has DIL produced
+   */
+  async hasDil(): Promise<boolean> {
+    return fs.pathExists(this.dilPath);
+  }
+
+  async ensureDilPath(qs: qsClient.QueryServerClient): Promise<string> {
```

Comment:
Can we unit test this? And also whether we always send `includeDilInQlo: true` in the compilation request (otherwise this command will not be very useful)?

Reply:
ok.